hibernate-core/src/main/java/org/hibernate/cfg/DependentSecondPass.java

```diff
@@ -0,0 +1,9 @@
+package org.hibernate.cfg;
+
+/**
+ * @author Matt Drees
+ */
+public interface DependentSecondPass extends SecondPass {
+
+	boolean dependentUpon(SecondPass secondPass);
```

Comment:
Coupled with my comment about storing these... I think this should accept `DependentSecondPass` arguments only.  And actually, since this only applies to `CopyIdentifierComponentSecondPass` I wonder if it might just make sense to use that for this contract.

I get that you are trying to make it generic.  But I'm not sure it gains us anything